Source/Controllers/Preferences/Panes/SPFilePreferencePane.m

```diff
@@ -178,56 +253,69 @@ - (void)updateHiddenFiles
 	[_currentFilePanel setShowsHiddenFiles:[prefs boolForKey:SPHiddenKeyFileVisibilityKey]];
 }
 
-- (void) chooseFile
-{
-	// retrieve the file manager in order to fetch the current user's home
-	// directory
-	NSFileManager *fileManager = [NSFileManager defaultManager];
-	NSURL *homeDirectory = nil;
-	if ([fileManager respondsToSelector:@selector(homeDirectoryForCurrentUser)]) {
-		homeDirectory = [fileManager homeDirectoryForCurrentUser];
-	} else {
-		homeDirectory = [NSURL fileURLWithPath:NSHomeDirectory()];
-	}
 
-	_currentFilePanel = [NSOpenPanel openPanel];
-	[_currentFilePanel setTitle:@"Choose ssh config"];
-	[_currentFilePanel setCanChooseFiles:YES];
-	[_currentFilePanel setCanChooseDirectories:YES];
-	[_currentFilePanel setAllowsMultipleSelection:YES];
-	[_currentFilePanel setAccessoryView:hiddenFileView];
-	[_currentFilePanel setResolvesAliases:NO];
-	[_currentFilePanel setDirectoryURL:[homeDirectory URLByAppendingPathComponent:@".ssh"]];
-	[self updateHiddenFiles];
-	
-	[prefs addObserver:self
-			forKeyPath:SPHiddenKeyFileVisibilityKey
-			   options:NSKeyValueObservingOptionNew
-			   context:NULL];
-	
-	[_currentFilePanel beginWithCompletionHandler:^(NSInteger returnCode)
-	 {
-		// only process data, when the user pressed ok
-		if (returnCode != NSModalResponseOK) {
-			return;
-		}
+- (void)chooseFileWithOptions:(PanelOptions)options
+{
+    // retrieve the file manager in order to fetch the current user's home
+    // directory
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+    NSURL *directory = nil;
+    if(options.fileName == nil){
+        if ([fileManager respondsToSelector:@selector(homeDirectoryForCurrentUser)]) {
+            directory = [[fileManager homeDirectoryForCurrentUser] URLByAppendingPathComponent:@".ssh"];
+        } else {
+            directory = [[NSURL fileURLWithPath:NSHomeDirectory()] URLByAppendingPathComponent:@".ssh"];
+        }
+    }
+    else{
+        NSString *staleFileDir = [options.fileName stringByDeletingLastPathComponent];
+        SPLog(@"staleFileDir: %@", staleFileDir);
+        directory = [NSURL fileURLWithPath:staleFileDir];
```

Comment:
It looks like there may be a way to pre-select the file even, by including the filename in the path (and possibly ending with a trailing /). Navigating to the directory automatically seems to be working great!
https://stackoverflow.com/questions/18903904/how-do-you-preselect-a-file-in-the-nsopenpanel-dialog